### DISCUSSION TITLE
Power and Heat

### DISCUSSION THREAD

**Discussion by kmycek** [tesla-android/issue-tracker#162] [Q&A]
So I am in the final throws of implementation. Printed a case and bought a good heatsink/fan. 

1) What's the recommended power source for the Pi? I have a 2022 M3LR, so I have USB-C (and a 12V cigarette lighter outlet) in the center console. All things I read say that they are USB-C 3.0 compliant, and should provide a maximum of 4.5 watts. Is that enough for the RPi 4 and CarlinKit dongle? I have USB-A in the glove box, but I would need a small hub, so I can continue to use the flash drive for Sentry/Dashcam. 

2) Heat - Where is a good place to place the single stack Pi in the car? I'm thinking that the glove box, or either of the two center consoles will be too restrictive with the airflow. I live in Arizona, so the inside of my car is always warm - well, at least 105 max due to Cabin Overheat Protection. Perhaps in the passenger footwell, tucked up out of view? In which location do people have the most success and least overheating issues. Besides slow performance - how will I know when the Pi is too hot?

**Comment by adamChrystie** [comment]
I have a 2022 Performance Model Y. I live in the hotter areas of Los Aneles due to being in a valley where ocean breeze is blocked mostly. I put the pi under the center arm rest where the cigarette lighter is. This area is not as tightly sealed as other areas so there will be less heat build up, feel with your fingers, there are open spaces. The Tesla owner manual (at least mine said) to not use usb bridges in the glovebox adit could interfere with dash cam . That being said people on forums say they have no issues but I didn't want to risk it.

I also have the [52Pi ice tower cooler fan](https://www.52pi.com/home/136-52pi-ice-tower-cpu-cooling-fan-ice-tower-fan-for-raspberry-pi-4-model-b-raspberry-pi-3b-raspberry-pi-3-model-b.html) since I live in hot California. There's also a lower profile version. These types of fan/heat syncs might not be needed as much since the most recent build of Tesla-android doesn't suggest overclocking anymore. But I like the steampunk appearance for now. I also have carlinkit dongle, a mini travel router + Alcatel LTE USB Modem stored under there and all powered vi the cigarette adapter. No issues in terms of space. This is the fairly open air acrylic transparent & black [case by 52Pi ](https://www.52pi.com/home/140-52pi-acrylic-transparent-clear-black-case-cover-for-raspberry-pi-4-model-b-with-cooling-fan-for-raspberry-pi-4b.html) that I use with the large fan/heat sync. I did not use the small fan that they provided. I do not get kickback from 52Pi.

I [have this Baseus 160W 3 port (two usb-c and one usb-a) cigarette adapter](https://www.amazon.com/gp/product/B09HTTSJLV/ref=ppx_yo_dt_b_asin_title_o04_s00?ie=UTF8&psc=1) and it powers my Pi and a small travel router device simultaneously. No issues. I tried cheaper micro adapter which looked like a good idea but it slips out easily and also doesn't block the plastic cover of the cigarette lighter port which wants to shut due to a spring. Go for a bunker longer design. 

Why not use the usb-c ports below center infotainment screen? IMHO, ports can be made more useful via updating them to support power & data operations. I can point you to a good link about that topic if interested.  I did it in January 2022. This allows you to plug in usb storage with high quality compressed mp3 or uncompressed Flac audio and playback via Tesla usb music player app. sSay goodbye to bluetooth compression! You can plug two game controllers such as PowerA in to  the usb-c via a usb-a to usb-c adapters and play some head to head games with family/friends.

**Comment by gem-404** [comment]
What's this

**Comment by mrnazu** [comment]
??

**Reply by iam-py-test** [reply]
Looks like a random spambot posting links to "meet-lonely-women-near-you"-type spam links, which pinged a ton of people and then deleted it's comment.

**Comment by gem-404** [comment]
okay...
